File: core/src/ru/myitschool/dcrawler/effects/MarkedEffect.java
package ru.myitschool.dcrawler.effects;

import com.badlogic.gdx.graphics.Texture;
import ru.myitschool.dcrawler.entities.Entity;

/**
 * Created by Voyager on 28.11.2017.
 */
public class MarkedEffect extends Effect {

    private String name = "Marked";
    private String description = "";
    private Texture icon = new Texture("mark.png");
    private boolean positive = false;
    private int penalty; //penalty is negative number
    private Entity marker;

    public MarkedEffect(Entity entity, Entity marker, int penalty) {
        super(entity);
        this.penalty = penalty;
        this.marker = marker;
    }

    @Override
    public int accuracyBonus(int accuracy, Entity target) {
        if (target != marker){
            accuracy += penalty;
        }
        return accuracy;
    }

    @Override
    public Texture getIcon() {
        return icon;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return "All your attacks except of attacks to creature that marked you gets " + penalty + " accuracy penalty.";
    }

    @Override
    public boolean isSkillUse() {
        return false;
    }

    @Override
    public boolean isPositive() {
        return false;
    }

    @Override
    public boolean isStackable() {
        return false;
    }

    @Override
    public int getStackSize() {
        return 0;
    }

    @Override
    public boolean isExpiring() {
        return false;
    }

    @Override
    public int getExpireTurns() {
        return 0;
    }

    @Override
    public String getType() {
        return "main.dcrawler.effect.marked";
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/MarkedEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.entities.Entity

/**
 * Created by Voyager on 28.11.2017.
 */
class MarkedEffect(entity: Entity?, private val marker: Entity?, //penalty is negative number
                   private val penalty: Int) : Effect(entity) {
    override val name = "Marked"
    override val description = "All your attacks except of attacks to creature that marked you gets $penalty accuracy penalty."
    override val icon = Texture("mark.png")
    override val positive: Boolean = false
    override val skillUse: Boolean = false
    override val stackable: Boolean = false
    override val stackSize: Int = 0
    override val expiring: Boolean = false
    override val expireTurns: Int = 0
    override val id: String = "main.dcrawler.effect.marked"

    override fun accuracyBonus(accuracy: Int, target: Entity): Int {
        var accuracy = accuracy
        if (target !== marker) {
            accuracy += penalty
        }
        return accuracy
    }
}